### DISCUSSION TITLE
[IOS] Notifications works well when App is foregrounded. But when backgrounded it doesnt work

### DISCUSSION THREAD

**Discussion by sagopaj** [thudugala/Plugin.LocalNotification#447] [Q&A]
I have a .net maui application which listens a postgresql database and when new row inserted to the database it raises a notification about this added row.

On android, it works well when app is foregrounded or backgrounded.

But on IOS it doesnt work when App is backgrounded. 

What should I do? Create a background service for listening that postgresql channel? 

Is some extra actions about background fetching or remote notifications required? I examined the source code and I think this plugin handles this issues itself.

At info.plist I checked following;
- Background fetch
- Background processing
- Remote notifications

And added platform/Ios the CustomUserNotificationCenterDelegate.cs class as instructed. Also added this to mauiprogram.cs
                .AddiOS(iOS =>
                {
#if IOS
                    iOS.SetCustomUserNotificationCenterDelegate(new CustomUserNotificationCenterDelegate());
#endif
                });

Thx.

**Comment by thudugala** [comment]
@sagopaj Can you please attach a sample project?
